### PR TITLE
devcontainer-cli: init at 0.53.0, nodePackages.node-pty: init at 1.0.0

### PR DESCRIPTION
**Pull request by h7x4**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/de/devcontainer-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  };
+
+  nativeBuildInputs  = [
+    makeWrapper
+    nodejs
+    prefetch-yarn-deps
+    yarn
+  ];
+
+  runtimeDependencies = [
```
Comment (SuperSandro2000) [inline pkgs/by-name/de/devcontainer-cli/package.nix:39]: Is that even being used?

File: pkgs/by-name/de/devcontainer-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    yarn --offline --production install
+
+    rm -rf node_modules/node-pty
+    ln -s ${nodePackages.node-pty}/lib/node_modules/node-pty node_modules/node-pty
```
Comment (SuperSandro2000) [inline pkgs/by-name/de/devcontainer-cli/package.nix:71]: What is the error when we don't do this?

File: pkgs/by-name/de/devcontainer-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  passthru.tests.version = testers.testVersion {
+    package = devcontainer-cli;
+    inherit version;
+  };
+
+  meta = with lib; {
+    description = "CLI tool for working with vscode devcontainers";
+    homepage = "https://github.com/devcontainers/cli";
+    changelog = let
+      versionWithoutDots = builtins.replaceStrings ["."] [""] version;
```
Comment (SuperSandro2000) [inline pkgs/by-name/de/devcontainer-cli/package.nix:91]: This is not being used.

File: pkgs/by-name/de/devcontainer-cli/package.nix [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    in "https://github.com/devcontainers/cli/blob/main/CHANGELOG.md#versionWithoutDots";
```
Comment (SuperSandro2000) [inline pkgs/by-name/de/devcontainer-cli/package.nix:92]:
```suggestion
    changelog = "https://github.com/devcontainers/cli/blob/main/CHANGELOG.md#${lib.replaceStrings ["."] [""] version}";
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please use buildNpmPackage for the new node package.

File: pkgs/by-name/de/devcontainer-cli/package.nix (hunk 3 above)
Comment (h7x4) [reply pkgs/by-name/de/devcontainer-cli/package.nix:91]: Slight mishap, forgot `${}`. I'll just inline it as in the suggestion below.

File: pkgs/by-name/de/devcontainer-cli/package.nix (hunk 1 above)
Comment (h7x4) [reply pkgs/by-name/de/devcontainer-cli/package.nix:39]: Yes, the whole tool relies on these two for fetching, building and running its containers. It does so in this manner:

- [example of `git` usage](https://github.com/devcontainers/cli/blob/34f35c2e424dd681842c0ed22d4c4f7769673496/build/hygiene.js#L227)
- [example of `docker` usage](https://github.com/devcontainers/cli/blob/34f35c2e424dd681842c0ed22d4c4f7769673496/src/spec-node/featuresCLI/testCommandImpl.ts#L89)

File: pkgs/by-name/de/devcontainer-cli/package.nix (hunk 2 above)
Comment (h7x4) [reply pkgs/by-name/de/devcontainer-cli/package.nix:71]: As far as I can remember, it builds just fine but fails during runtime due to missing `../build/{Release,Debug}/pty.node`. The reason was that it's missing executables that were supposed to be built with node-gyp. That is the reason why I ended up packaging this package standalone as well.

**Comment by h7x4** [comment]
> Please use buildNpmPackage for the new node package.

Would you mind elaborating on this? Are packages no longer supposed to go into `pkgs/development/node-packages`?

File: pkgs/by-name/de/devcontainer-cli/package.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/by-name/de/devcontainer-cli/package.nix:39]: I think stdenv.mkDerivation is not using this by default.

**Comment by SuperSandro2000** [comment]
> Are these kinds of development node packages no longer supposed to go into `pkgs/development/node-packages`?

Yes, because it is a giant mess and very hard to maintain and generate.

**Comment by h7x4** [comment]
@mr360 

> Would you like to combine forces as we are essentially tackling the same problem :-)

Sure :)

**Pull request closed without merging**